### PR TITLE
New table "dcontact" for diaspora contacts

### PR DESCRIPTION
**Pull request by annando**
This table will act as a replacement for the `fcontact` table. Just like the `apcontact` table it contains data that is specific for the protocol.

We have to decide how to continue with this PR. We use it as an example for this new paradigm thingy. Possibly @MrPetovan could take over and use this as a starting point.

### CODE REVIEW

**Comment by tobiasd** [comment]
Is this meant to be in 2021.09 or 2021.12?

**Comment by annando** [comment]
This depends upon what @MrPetovan thinks about it. When we will perform the quick way then I don't see a problem for the RC. The other way could be more demanding with more changes to the core.

**Comment by MrPetovan** [comment]
I'd prefer it to be a stepping stone towards the new paradigm and it isn't crucial for this release, let's push it to the next.

File: src/Network/Probe.php [hunk 1]
```diff
@@ -738,19 +739,19 @@ private static function detect(string $uri, string $network, int $uid, array $ap
 
 		$result = [];
 
-		if (in_array($network, ["", Protocol::DFRN])) {
+		if (in_array($network, ['', Protocol::DFRN])) {
 			$result = self::dfrn($webfinger);
 		}
-		if ((!$result && ($network == "")) || ($network == Protocol::DIASPORA)) {
+		if (!$result && in_array($network, ['', Protocol::DIASPORA])) {
```
Comment (MrPetovan) [inline src/Network/Probe.php:745]: The new condition isn't equivalent to the previous one, is it intended?

With:
- `A`: `$result`
- `B`: `$network == ''`
- `C`: `$network == Protocol::DIASPORA`

Former condition: `(NOT(A) AND B) OR C`
New condition: `NOT(A) AND (B OR C)`

Comment (annando) [reply src/Network/Probe.php:745]: It doesn't matter here. I'm on my mobile, so I keep the answer short.

Comment (MrPetovan) [reply src/Network/Probe.php:745]: Thanks, I just wanted to make sure.

Comment (annando) [reply src/Network/Probe.php:745]: `$result` is empty both when the previous check wasn't successful (which would only be executed when the network had been empty) or when we would check for a dedicated network.

**Comment by MrPetovan** [comment]
I guess this will need to be merged after the release, huh? 😉

**Comment by annando** [comment]
@MrPetovan question is what to do with it. You preferred to have it in a new class structure. But I really feel unable doing so. You gave your best explaing the class stuff, but I'm missing basics here. So we will have to decide how to continue.

**Comment by MrPetovan** [comment]
I can take over the branch and apply the class structure to it for the next milestone. I started converting the notifications to the new paradigm, I have the `notify` table ready, but it’s proving longer than expected for `notifications` and `introductions` because of the included refactoring of `enotify.php`

**Comment by tobiasd** [comment]
This PR was closed, when the 2021.09-rc branch was deleted, is that correct? Or should it be re-opened and pointing towards deelop?

**Comment by MrPetovan** [comment]
The automatic closing of all the PRs targeting the `2021.09-rc` branch when it was deleted was inconvenient, but @annando and I were able to recreate all of them.

A better way would have been to change their target merge branch to `develop` before deleting the `2021.09-rc` branch but it is slightly more involved since you shouldn't be the one doing it on your own.

**Pull request closed without merging**